x-pack/plugins/security_solution/public/detections/components/rules/select_rule_type/index.tsx

```diff
@@ -47,54 +47,54 @@ export const SelectRuleType: React.FC<SelectRuleTypeProps> = ({
   const setQuery = useCallback(() => setType('query'), [setType]);
   const setThreshold = useCallback(() => setType('threshold'), [setType]);
   const setThreatMatch = useCallback(() => setType('threat_match'), [setType]);
-  const mlCardDisabled = isReadOnly || !hasValidLicense || !isMlAdmin;
   const licensingUrl = useKibana().services.application.getUrlForApp('kibana', {
     path: '#/management/stack/license_management',
   });
 
   const eqlSelectableConfig = useMemo(
```

Comment:
@FrankHassanabad it looks like these should've been explicitly typed as `EuiCardSelectProps`:
```suggestion
  const eqlSelectableConfig: EuiCardSelectProps = useMemo(
```

While this doesn't behave exactly as I would expect (you can still pass it to the component), you at least get a type error when trying to reference the `isVisible` field, which does not exist on that type. This is why you're getting those react warnings in your tests (and likely in the browser as well).

Reply:
It's not technically that all the properties are being used within the EuiCard. For example the `isVisible` is only being used as a boolean to determine if the card should show up or not:

```ts
{querySelectableConfig.isVisible && (
          <EuiFlexItem>
            <EuiCard
              data-test-subj="customRuleType"
              title={i18n.QUERY_TYPE_TITLE}
              titleSize="xs"
              description={i18n.QUERY_TYPE_DESCRIPTION}
              icon={<EuiIcon size="xl" type="search" />}
              selectable={querySelectableConfig}
              layout="horizontal"
              textAlign="left"
            />
          </EuiFlexItem>
        )}
```

The errors I am getting look to be from within `isVisible` being set by something else under the covers within Eui and not this `isVisible` I use here (unless I'm missing something ... which I could be :-)).